base/src/commonTest/kotlin/monitor/ServiceMonitorRepoTest.kt

```diff
@@ -0,0 +1,183 @@
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package monitor
+
+import com.splendo.kaluga.base.ServiceMonitor
+import com.splendo.kaluga.base.monitor.ServiceMonitorState
+import com.splendo.kaluga.base.monitor.ServiceMonitorStateRepo
+import com.splendo.kaluga.base.runBlocking
+import com.splendo.kaluga.base.utils.EmptyCompletableDeferred
+import com.splendo.kaluga.base.utils.complete
+import com.splendo.kaluga.test.BaseTest
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class ServiceMonitorRepoTest : BaseTest() {
+
+    companion object {
+        const val SERVICE_MONITOR_TIMEOUT = 200L
+    }
+
+    private val testCoroutine = SupervisorJob()
+    private val coroutineScope = CoroutineScope(Dispatchers.Main + testCoroutine)
+
+    private val stateHolder: MutableStateFlow<ServiceMonitorState?> = MutableStateFlow(null)
+
+    internal class StubMonitor(initialValue: Boolean = false) : ServiceMonitor {
+
+        var output = MutableStateFlow(initialValue)
+        val startMonitoringDeferred = EmptyCompletableDeferred()
+        val stopMonitoringDeferred = EmptyCompletableDeferred()
+
+        override val isServiceEnabled: Boolean = output.value
+        override val isEnabled: Flow<Boolean> = output
+
+        override fun startMonitoring() {
+            println("startMonitoring started")
+            startMonitoringDeferred.complete()
+        }
+
+        override fun stopMonitoring() {
+            stopMonitoringDeferred.complete()
+        }
+    }
+
+    @Test
+    fun test_first_state_is_not_initialized() = runBlockingWithStubMonitor(StubMonitor(false)) { _, monitorRepo ->
+        val firstValue = monitorRepo.first()
+        assertTrue { firstValue is ServiceMonitorState.NotInitialized }
+    }
+
+    @Test
+    fun test_init_state_is_initialized_enabled() = runBlockingWithStubMonitor(StubMonitor(true)) { _, monitorRepo ->
+        val job = launch { monitorRepo.collect() }
+        delay(SERVICE_MONITOR_TIMEOUT)
+        monitorRepo.useState {
+            assertTrue { it is ServiceMonitorState.Initialized.Enabled }
```

Comment:
just an idea, helper method in `test-utils` for this? `assertStateEquals` / `assertStateTrue` / `assertStateFalse`

Reply:
I cannot find any of those :O

Reply:
suggestion is to create them